tests/key_overrides/mod_neutralizer/test_key_overrides.cpp

```diff
@@ -0,0 +1,432 @@
+#include "test_driver.hpp"
+#include "test_fixture.hpp"
+#include "test_keymap_key.hpp"
+
+using testing::_;
+using testing::InSequence;
+
+class NeutralizedKeyOverrides : public TestFixture {};
+
+TEST_F(NeutralizedKeyOverrides, modifier_is_suppressed_but_not_neutralized) {
+    // Tested key override:
+    // ko_make_basic(MOD_MASK_SHIFT, KC_BSPC, KC_DEL),
+    //
+    // MOD_MASK_SHIFT is not part of MODS_TO_NEUTRALIZE so we need to make
+    // sure that no dummy mod neutralizer keycode gets sent when this key
+    // override is getting triggered. However, the replacement (KC_DEL) is
+    // not a shifted keycode, so the shift of the trigger should be
+    // suppressed nonetheless.
+
+    TestDriver driver;
+    KeymapKey  key_lctrl(0, 0, 0, KC_LEFT_CTRL);
+    KeymapKey  key_lsft(0, 0, 1, KC_LEFT_SHIFT);
+    KeymapKey  key_bspc(0, 1, 0, KC_BACKSPACE);
+
+    set_keymap({key_lctrl, key_lsft, key_bspc});
+
+    key_lsft.press();
+    EXPECT_REPORT(driver, (KC_LEFT_SHIFT));
+    run_one_scan_loop();
+    VERIFY_AND_CLEAR(driver);
+
+    // Left shift should be removed from the report but no dummy keycode
+    // should be inserted because left shift is not in the list of mods
+    // to neutralize.
+    key_bspc.press();
+    EXPECT_REPORT(driver, (KC_DELETE));
+    run_one_scan_loop();
+    VERIFY_AND_CLEAR(driver);
+
+    // Left shift is still held down so we should restore it after
+    // sending the overriden key with suppressed mods.
+    key_bspc.release();
+    EXPECT_REPORT(driver, (KC_LEFT_SHIFT));
+    run_one_scan_loop();
+    VERIFY_AND_CLEAR(driver);
+
+    key_lsft.release();
+    EXPECT_EMPTY_REPORT(driver);
+    run_one_scan_loop();
+    VERIFY_AND_CLEAR(driver);
+}
+
+TEST_F(NeutralizedKeyOverrides, modifier_is_suppressed_and_neutralized) {
+    // Tested key override:
+    // ko_make_basic(MOD_BIT_LGUI, KC_H, KC_6)
+    //
+    // MOD_BIT_LGUI is part of the MODS_TO_NEUTRALIZE so the
+    // DUMMY_MOD_NEUTRALIZER_KEYCODE should get sent during
+    // the suppression of the GUI modifier.
+
+    TestDriver driver;
+    KeymapKey  key_lgui(0, 0, 0, KC_LEFT_GUI);
+    KeymapKey  key_h(0, 1, 0, KC_H);
+
+    set_keymap({key_lgui, key_h});
+
+    key_lgui.press();
+    EXPECT_REPORT(driver, (KC_LEFT_GUI));
+    run_one_scan_loop();
+    VERIFY_AND_CLEAR(driver);
+
+    key_h.press();
+    {
+        InSequence seq;
+
+        EXPECT_REPORT(driver, (KC_LEFT_GUI, DUMMY_MOD_NEUTRALIZER_KEYCODE));
+        EXPECT_REPORT(driver, (KC_LEFT_GUI));
+        EXPECT_REPORT(driver, (KC_6));
+    }
+    run_one_scan_loop();
+    VERIFY_AND_CLEAR(driver);
+
+    key_h.release();
+    EXPECT_REPORT(driver, (KC_LEFT_GUI));
+    run_one_scan_loop();
+    VERIFY_AND_CLEAR(driver);
+
+    key_lgui.release();
+    // The two following reports should also be sent but they're not.
+    // See qmk_firmware issue #22208.
+    // EXPECT_REPORT(driver, (KC_LEFT_GUI, DUMMY_MOD_NEUTRALIZER_KEYCODE));
+    // EXPECT_REPORT(driver, (KC_LEFT_GUI));
```

Comment:
I wasn't sure on what was the best way to add those “wishful” assertions. I know there are (were?) other “wishful” unit tests that highlighted a current bug and wouldn't pass that were configured to be skipped but I didn't like the idea of skipping this whole test due to a single part of it.